sdks/python/apache_beam/io/textio.py

```diff
@@ -86,7 +90,9 @@ def __init__(self,
                coder,
                buffer_size=DEFAULT_READ_BUFFER_SIZE,
                validate=True,
-               skip_header_lines=0):
+               skip_header_lines=0,
+               header_matcher_predicate=None,
+               header_processor=None):
```

Comment:
<!--new_thread; commit:70ade9d4fe6a8e31ff0be4d39b731510c93b4bf9; resolved:0-->
This API is a bit confusing since 'header_processor' is only used if "header_matcher_predicate" is provided not when "skip_header_lines" is provided. How about combining inputs predicate and processor to a single tuple ?